src/doom/g_game.c

```diff
@@ -304,6 +305,10 @@ static int G_NextWeapon(int direction)
         }
     }
 
+    // The current weapon must be present in weapon_order_table
+    // otherwise something has gone terribly wrong
+    assert(i < arrlen(weapon_order_table));
```

Comment:
That'd be the only place where we use `assert()` in actual game code. Not sure about it. @fragglet ?

Reply:
That depends on how we define "game code". There are already asserts in `i_main.c`, `i_sdlsound.c`, `m_config.c`, `midifile.c` and the OPL code.